vizro-core/tests/unit/vizro/models/test_models_utils.py

```diff
@@ -21,3 +22,37 @@ def test_check_for_valid_component_types(self, model_with_layout):
             match=re.escape("(allowed values: 'ag_grid', 'button', 'card', 'container', 'graph', 'table', 'tabs')"),
         ):
             model_with_layout(title="Page Title", components=[vm.Checklist()])
+
+
+@pytest.mark.parametrize(
+    "test_string, expected",
+    [
+        ("Title", "title"),
+        ("Page Title", "page-title"),
+        ("Page Title!", "page-title"),
+        ("Another123Title", "another123title"),
+        ("Some_Example%Title", "some_exampletitle"),
+        ("!@#$%^&*()_+=", "_"),
+        ("", ""),
+    ],
+)
+def test_clean_path_with_url_enabled(test_string, expected):
+    cleaned_string = _clean_path(path=test_string, allowed_characters="-_/", enable_url=True)
+    assert cleaned_string == f"/{expected}"
+
+
+@pytest.mark.parametrize(
+    "test_string, expected",
+    [
+        ("Title", "title"),
+        ("Page Title", "page-title"),
+        ("Page Title!", "page-title"),
+        ("Another123Title", "another123title"),
+        ("Some_Example%Title", "some_exampletitle"),
+        ("!@#$%^&*()_+=", "_"),
+        ("", ""),
+    ],
+)
+def test_clean_path_with_url_disabled(test_string, expected):
+    cleaned_string = _clean_path(path=test_string, allowed_characters="-_/", enable_url=False)
+    assert cleaned_string == expected
```

Comment:
Was opening an issue just now and randomly came across this :) You can do multiple `parametrize` on the same test, which is like a nested for loop, so those two tests can be joined into one if you like.

```suggestion
@pytest.mark.parametrize("enable_url", [True, False])
@pytest.mark.parametrize(
    "test_string, expected",
    [
        ("Title", "title"),
        ("Page Title", "page-title"),
        ("Page Title!", "page-title"),
        ("Another123Title", "another123title"),
        ("Some_Example%Title", "some_exampletitle"),
        ("!@#$%^&*()_+=", "_"),
        ("", ""),
    ],
)
def test_clean_path_with_url_enabled(test_string, expected, enable_url):
    cleaned_string = _clean_path(path=test_string, allowed_characters="-_/", enable_url=enable_url)
    expected_output = f"/{expected}" if enable_url else expected
    assert cleaned_string == expected_output
```

Reply:
Haha, amazing! 🚀 I actually parameterised all four arguments before the way you proposed it above, but I thought it wasn't easy to read. I like your suggestion above though, as it is still kept very simple! Thanks!!

Reply:
Welcome. Also might be helpful to have type hints in test functions signatures too (that would show that last param is `bool` while two others are `str`, which is probably what you meant by "difficult to read").

But I didn't add it in my example to keep it isolated to the double parametrization only.

Reply:
That's a great suggestion 👍 We do the type hints on the actual functions we test so we have it there, but I don't think we do it on the test function signatures yet. 

I had this before and then a long list of use cases 😄 In the end I thought, it became more difficult to read and understand because in the end `enable_url=True` just adds a `/` at the beginning, but that didn't become immediately clear from screening through all the use cases. Didn't think about just adding a simple if/else - haha! Thanks for the suggestion and your first contribution! 🚀 

```
@pytest.mark.parametrize(
    "test_string, allowed_characters, enable_url, expected",
    [
        ("Title", "-_/", True, "/title"),
        ("Title", "-_/", False, "title"),
        ...
    ],
)
def test_clean_path_with_url_enabled(test_string, allowed_characters, enable_url, expected):
    cleaned_string = _clean_path(path=test_string, allowed_characters=allowed_characters, enable_url=enable_url)
    assert cleaned_string == expected_output
```